### PR TITLE
Add uClibc definitions to core.sys.posix.*

### PR DESCRIPTION
**Pull request by rracariu**

### CODE REVIEW

File: src/core/sys/posix/fcntl.d [hunk 1]
```diff
@@ -844,6 +844,126 @@ else version( CRuntime_Musl )
...
+    enum F_GETFL        = 3;
+    enum F_SETFL        = 4;
+
+    version(X86_64)
+    {
+        enum F_GETLK        = 5;
+        enum F_SETLK        = 6;
+        enum F_SETLKW       = 7;
+    }
+    else static if (__USE_FILE_OFFSET64)
```
Comment (joakim-noah) [inline src/core/sys/posix/fcntl.d]: Didn't think you could mix `version` and `static if` like this, does it actually work?

File: src/core/sys/posix/stdio.d [hunk 2]
```diff
@@ -217,12 +273,16 @@ else version( DragonFlyBSD )                 // for DragonFlyBSD
...
+else version( CRuntime_UClibc )
+    version = HaveMemstream;
 
 version( HaveMemstream )
 {
     FILE*  fmemopen(in void* buf, in size_t size, in char* mode);
     FILE*  open_memstream(char** ptr, size_t* sizeloc);
-    FILE*  open_wmemstream(wchar_t** ptr, size_t* sizeloc);
+    version( CRuntime_UClibc ) {}
+    else
```
Comment (joakim-noah) [inline src/core/sys/posix/stdio.d]: Can just put the `else` at the end of the line above and not re-indent the next line.

File: src/core/sys/posix/stdio.d [hunk 3]
```diff
@@ -343,6 +417,8 @@ unittest
     assert(fclose(f) == 0);
 }
 
+version( CRuntime_UClibc ) {}
+else
```
Comment (joakim-noah) [inline src/core/sys/posix/stdio.d]: Same here.

File: src/core/sys/posix/sys/types.d [hunk 4]
```diff
@@ -1022,6 +1077,150 @@ else version( CRuntime_Bionic )
...
+        enum __SIZEOF_PTHREAD_MUTEX_T = 24;
+        enum __SIZEOF_PTHREAD_MUTEXATTR_T = 4;
+        enum __SIZEOF_PTHREAD_COND_T = 48;
+        enum __SIZEOF_PTHREAD_COND_COMPAT_T = 12;
+        enum __SIZEOF_PTHREAD_CONDATTR_T = 4;
+        enum __SIZEOF_PTHREAD_RWLOCK_T = 32;
+        enum __SIZEOF_PTHREAD_RWLOCKATTR_T = 8;
+        enum __SIZEOF_PTHREAD_BARRIER_T = 20;
+        enum __SIZEOF_PTHREAD_BARRIERATTR_T = 4;
+     }
```
Comment (joakim-noah) [inline src/core/sys/posix/sys/types.d]: Might help to throw a static assert here if the architecture isn't found.

File: src/core/sys/posix/sys/types.d [hunk 5]
```diff
@@ -1090,6 +1289,21 @@ else version( CRuntime_Bionic )
 else version( CRuntime_Musl )
 {
 }
+
```
Comment (joakim-noah) [inline src/core/sys/posix/sys/types.d]: Don't add blank line.

File: src/core/sys/posix/sys/types.d [hunk 6]
```diff
@@ -1122,7 +1336,10 @@ else version (Solaris)
 {
     alias pthread_mutex_t pthread_spinlock_t;
 }
-
```
Comment (joakim-noah) [inline src/core/sys/posix/sys/types.d]: Leave blank line below.

File: src/core/sys/posix/fcntl.d (hunk 1 above)
Comment (rracariu) [reply src/core/sys/posix/fcntl.d]: Yes, it works. It is also used on the Glibc version, see line 102

File: src/core/sys/posix/sys/types.d (hunk 4 above)
Comment (rracariu) [reply src/core/sys/posix/sys/types.d]: OK, added.

**Comment by rracariu** [comment]
Fixed the style and formatting issues.

**Comment by rracariu** [comment]
Thanks!

**Pull request merged**